manifests/keyring.pp

```diff
@@ -0,0 +1,65 @@
+# @summary Manage GPG keyrings for apt repositories
+#
+# @example Install the puppetlabs apt source with keyring.
+#     apt::source { 'puppet7-release':
+#       location => 'http://apt.puppetlabs.com',
+#       repos    => 'main',
+#       keyring  => '/etc/apt/keyrings/puppetlabs-keyring.gpg',
+#     }
+#     apt::keyring {'puppetlabs-keyring.gpg':
+#       source => 'https://apt.puppetlabs.com/keyring.gpg',
+#     }
+#
+# @param keyring_dir
+#   Path to the directory where the keyring will be stored.
+#
+# @param keyring_filename
+#   Optional filename for the keyring.
+#
+# @param keyring_file
+#   File path of the keyring.
+#
+# @param keyring_file_mode
+#   File permissions of the keyring.
+#
+# @param source
+#   Source of the keyring file. Mutually exclusive with 'content'.
+#
+# @param content
+#   Content of the keyring file. Mutually exclusive with 'source'.
+#
+# @param ensure
+#   Ensure presence or absence of the resource.
+#
+define apt::keyring (
+  Stdlib::Absolutepath $keyring_dir = '/etc/apt/keyrings',
```

Comment:
This is already defined as `apt::params::trusted_gpg_d`

Reply:
True, but for new-style APT keyrings, `/etc/apt/keyrings` is the preferred location.

Changing it only here means the new-style keyrings can be deployed to `/etc/apt/keyrings` by default, without affecting the rest of the module.